config/routes.rb

```diff
@@ -1,50 +1,49 @@
+  get "manage/render_admin_menue"      => "manage#render_admin_menue"
+  get "call_for_support"               => "manage#call_for_support"
+  post "frontend_login/:usermodel"     => "sessions#login", as: :frontend_login
+  post "frontend_register/:usermodel"  => "sessions#register", as: :frontend_register
+  post "manage/article_visibility/:id" => "manage#article_visibility"
+
+  mount Sidekiq::Web => "/admin/background" if RUBY_VERSION.to_f >= 1.9
+
+  # post "/api/v1/tokens" => "goldencobra/api/v1/tokens_controller#create"
+  namespace "api" do
+    namespace "v1" do
       resources :tokens, only: [:create, :show]
     end
 
-    namespace 'v2' do
-      get '/articles'                 => 'articles#index'
-      get '/articles/search'          => 'articles#search'
-      get '/articles/breadcrumb/*url' => 'articles#breadcrumb'
-      get '/articles/*url'            => 'articles#show'
-      get '/locale_string'            => 'locales#get_string'
-      get '/setting_string'           => 'settings#get_string'
-      get '/uploads'                  => 'uploads#index'
-      post '/articles/create'         => 'articles#create'
-      post '/articles/update'         => 'articles#update'
-      get '/navigation_menus'         => 'navigation_menus#index'
-      get '/navigation_menus/active'  => 'navigation_menus#active_ids'
+    namespace "v2" do
+      get "/articles"                 => "articles#index"
+      get "/articles/search"          => "articles#search"
+      get "/articles/breadcrumb/*url" => "articles#breadcrumb"
+      get "/articles/*url"            => "articles#show"
+      get "/locale_string"            => "locales#get_string"
+      get "/setting_string"           => "settings#get_string"
+      get "/uploads"                  => "uploads#index"
+      post "/articles/create"         => "articles#create"
+      post "/articles/update"         => "articles#update"
+      get "/navigation_menus"         => "navigation_menus#index"
+      get "/navigation_menus/active"  => "navigation_menus#active_ids"
     end
   end
 
-  get 'sitemap', to: 'articles#sitemap', defaults: { format: 'xml' }
+  get "sitemap", to: "articles#sitemap", defaults: { format: "xml" }
 
-  devise_for :visitors, controllers: { omniauth_callbacks: 'visitors/omniauth_callbacks' }
+  devise_for :visitors, controllers: { omniauth_callbacks: "visitors/omniauth_callbacks" }
   devise_scope :visitors do
-    get '/visitors/auth/:provider' => 'omniauth_callbacks#passthru'
+    get "/visitors/auth/:provider" => "omniauth_callbacks#passthru"
   end
 
-  #match '/*article_id.pdf', to: 'articles#convert_to_pdf'
-  get '/*article_id', to: 'articles#show'
+  # match "/*article_id.pdf", to: "articles#convert_to_pdf"
+  get "/*article_id", to: "articles#show", constraints: lambda { |req|
+    # php requests results in MimeType:Null-Object Github #47
+    !["php", ""].include?(req.format.to_sym.to_s.downcase)
```

Comment:
I would refrain from using the settings, as every request would lead to a database request, right?
Perhaps better use a config file in a project?